internal/controllers/watch/kind.go

```diff
@@ -125,30 +126,37 @@ func (k *KindWatchController) newResourceWatchController(parent *WatchController
 // buildRequests returns a reconcile request for every binding to this resource kind.
 func (k *KindWatchController) buildRequests(synth *apiv1.Synthesizer, comps ...apiv1.Composition) []reconcile.Request {
 	keys := map[string]struct{}{}
+	reqs := []reconcile.Request{}
 	for _, ref := range synth.Spec.Refs {
 		keys[ref.Key] = struct{}{}
+
+		if ref.Resource.Name == "" {
+			continue // ref does not have an "implicit" binding
+		}
+
+		nsn := types.NamespacedName{Namespace: ref.Resource.Namespace, Name: ref.Resource.Name}
+		req := reconcile.Request{NamespacedName: nsn}
+		if !slices.Contains(reqs, req) {
+			reqs = append(reqs, req)
+		}
 	}
 
-	reqs := []reconcile.Request{}
 	for _, comp := range comps {
 		for _, binding := range comp.Spec.Bindings {
 			if _, found := keys[binding.Key]; !found {
 				continue
 			}
 
 			nsn := types.NamespacedName{Namespace: binding.Resource.Namespace, Name: binding.Resource.Name}
-			var exists bool
-			for _, req := range reqs {
-				if req.NamespacedName == nsn {
-					exists = true
-					break
-				}
-			}
-			if !exists {
-				reqs = append(reqs, reconcile.Request{NamespacedName: nsn})
+			req := reconcile.Request{NamespacedName: nsn}
```

Comment:
The new change is updating the existing request with the name/namespace defined in comp binding, and append new request if not found? If this is correct, the logic should be:
```go
idx := slices.Index(reqs, req)
if idx >= 0 {   // >= 0
  reqs[idx] = req  // update existing request but don't need to append it
} else {
  reqs = append(reqs, req) // append request for the new request only
}
```

Reply:
Pushed a commit to fix/clarify the logic here. Implicit bindings currently take precedence over explicit so the synth will win. I _think_ this makes sense - if the synth "knows" a resource it needs, the composition shouldn't be able to say otherwise. But open to other ideas.

Reply:
I rationalized it backwards. For me the Synthesizer is a default because it's the "template" and the Composition being the instantiation is more specific and should take precedence.

Reply:
I don't think the specific behavior is important though, as long as it's consistent and called out